### DISCUSSION TITLE
Open Multiple Categories for Discussions

### DISCUSSION THREAD

**Discussion by sarthakgaur** [directus/directus#19462] [Feature Requests]
### Summary

I just wrote down how to downgrade directus in an issue thread. Things like this can be shared in a discussion thread. Search engines can index these discussions. And, it can help a few people out.

https://github.com/directus/directus/issues/19327#issuecomment-1682146040 

### Basic Example

Check summary.

### Motivation

Important things can be shared in these discussion threads that can be indexed by search engines. It doesn't have to be just about feature requests. It can become a great way to get help from the community. This is a lot better than using Discord. Discord discussions cannot be indexed by search engines. It be great if people move away from Discord and start using these Github features.

### Detailed Design

N/A.

### Requirements List

N/A.

### Drawbacks

There is really no reason to not do this. You had multiple categories before. I am not sure why they were removed.

### Alternatives

N/A.

### Adoption Strategy

N/A.

**Comment by rijkvanzanten** [comment]
> There is really no reason to not do this. You had multiple categories before. I am not sure why they were removed.

Because it was get very difficult to triage and organize questions intertwined with issues and discussions in both Discord and GitHub. People were way more active in helping each other on Discord so we decided it was the better place to funnel all questions to 🤔

**Reply by sarthakgaur** [reply]
@rijkvanzanten [Discord is a black hole for information](https://news.ycombinator.com/item?id=30311982).

**Reply by rijkvanzanten** [reply]
🤷🏻 a thread of opinions on hacker news is not really a source I can work with..

**Reply by sarthakgaur** [reply]
Okay, @rijkvanzanten. I'll write down why Discord is a wrong platform for these type of conversations. Although, I would highly suggest going through the discussion on Hacker News and [here](https://knockout.chat/thread/33251/1). They go into great detail.

1. Search engines cannot index these discussions on Discord. I cannot stress the importance of this enough. This alone should discourage people from having these discussions on Discord.
2. The search functionality of Discord is subpar to say the least.
3. Hard to refer to previous discussions regarding the same topic.
4. Hard to share Discord messages outside of Discord. If a member of your team is not on Discord and you gave them a Discord "Message Link". They'll first have to create an account on Discord, join the same server, then use that link to see the message. None of this is a problem with GitHub discussions. You just give them a link. They don't even need a GitHub account to view the discussion thread.
5. Discord messages should be under a certain character limit. You can increase that limit slightly by subscribing to Discord Nitro. But, the limit is still short. You can get around it by sending text files. None of this is really an issue on GitHub discussions.
6. All the sorting, labeling, and filtering features are not as good as GitHub discussions.
7. You can only have a limited number of active forum posts or threads on Discord.

I have said it in the past and I'll say it again. Discord is not an alternative to forums. It's a chatting platform built to replace software like Mumble, Ventrilo, TeamSpeak, etc. Discord does support forums, but it pales in comparison. It's absolutely beyond me why open source projects feel the need to have a presence on Discord. The community is already there on GitHub. Why recreate this community which contains essentially the same bunch of people on another platform? And, it's even more frustrating when Discord is suggested as the primary way to get help.

Discord makes sense for certain communities. But, if the goal is to create a repository of information that is easily accessible, it fails. And, it fails catastrophically. Discord truly is a black hole for information. If anything goes in, it's not coming out.

> Because it was get very difficult to triage and organize questions intertwined with issues and discussions in both Discord and GitHub. People were way more active in helping each other on Discord so we decided it was the better place to funnel all questions to 🤔

I know not many people will like this, but the right move was to move away from Discord. But, since you guys are now heavily involved with Discord, it will be a painful process to migrate everything to GitHub. But, it's still something to seriously consider.

**Reply by phazonoverload** [reply]
I run the community with Rijk, and I'm very conflicted. It wouldn't be too much work to get general discussion threads going here, and I'm tentatively up for it, but we were finding that folks would open threads in both places and it just became a mess - having a mutually-exclusive set of uses for different platforms has helped us in that regard.

**Reply by Pachat** [reply]
The question is indeed to what extent the community wants to be self-centered or willing to spread the excellence of Directus widely across the world.

**Reply by phazonoverload** [reply]
What do you mean?

**Reply by Pachat** [reply]
I mean, building brand awareness is an ongoing, long-term effort. 
I really appreciate the great initiatives that have taken place from Directus towards the community this year :)
They deserve to be known: “Discussions” are therefore a good SEO opportunity not to be overlooked.

**Reply by rijkvanzanten** [reply]
True true! Although a web-facing index of the questions asked in Discord could be an SEO solution as well, kinda like what Nextjs started doing: https://nextjs-forum.com/ 🤔 I'm more curious around the UX of opening up gh discussions for questions vs discord. In my experience, people are more likely to help each other on discord than on GH

**Reply by Pachat** [reply]
>index of questions asked in Discord could also be an SEO solution

Certainly. This would allow a smarter search than in Discord

On the UX side, my preference is clearly towards GH. But that's just a personal opinion.

**Reply by abdonrd** [reply]
My two cents is that I prefer GitHub Discussions over Discord.

I find it much more comfortable to link other issues/PRs, etc. Even the source code.

I have had very good discussions here like:
- https://github.com/directus/directus/discussions/11798
(notice that I just paste the link, and GitHub shows the title of the discussion, with a detailed tooltip on hover)

And the GitHub search seems very good to me, I always end up finding what I'm looking for.

I find Discord comfortable only for quick, real-time conversations.

**Reply by sarthakgaur** [reply]
> I run the community with Rijk, and I'm very conflicted. It wouldn't be too much work to get general discussion threads going here, and I'm tentatively up for it, but we were finding that folks would open threads in both places and it just became a mess - having a mutually-exclusive set of uses for different platforms has helped us in that regard.

@phazonoverload Archive all the help posts on Discord. Don't let people create new help posts on Discord, point them to GitHub discussions. If you want to have your brand presence on Discord, use it exclusively for live chat and voice chat.

> True true! Although a web-facing index of the questions asked in Discord could be an SEO solution as well, kinda like what Nextjs started doing: https://nextjs-forum.com/

@rijkvanzanten They are building a solution to a problem that can be completely avoided in the first place.

> I'm more curious around the UX of opening up gh discussions for questions vs discord. In my experience, people are more likely to help each other on discord than on GH

@rijkvanzanten If you weigh the pros and cons of both Discord and GitHub discussions, GitHub discussions is the clear winner.

**Reply by rijkvanzanten** [reply]
> Archive all the help posts on Discord. Don't let people create new help posts on Discord, point them to GitHub discussions. If you want to have your brand presence on Discord, use it exclusively for live chat and voice chat.

This won't work. People will ask questions there regardless. The same goes for stackoverflow/reddit/twitter etc. If we have a presence on the platform people will use it to ask questions. The discussion here is to prioritize just Discord, or both Discord & GH.

> They are building a solution to a problem that can be completely avoided in the first place.

I don't think it's that black and white.

> If you weigh the pros and cons of both Discord and GitHub discussions, GitHub discussions is the clear winner.

Maybe from the outside looking in, but don't agree at all it's that obvious of a decision.

**Comment by Pachat** [comment]
My experience is that I found out about the existence of Directus through a Google search which led me to a discussion in another headless cms.
Just to give some thoughts, so far, "strapi headless" brings up 126,000 results on Google, "directus headless" 31,200.
Any referral source for Directus should be considered.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
